src/main.ts

```diff
@@ -13,17 +13,24 @@ if (selection.length !== 1 || !isShape(node)) {
   figma.closePlugin();
 }
 
-let fillIndex = getFillIndexWithGradient(node);
-if (fillIndex === null) {
+if (getFillIndexWithGradient(node) === null) {
   figma.notify("Please select a shape with a gradient");
   figma.closePlugin();
 }
 
-const fill = node.fills[fillIndex];
-console.log("Fill:", fill);
+let fills = node.fills as readonly Paint[];
 
-const newColors = easeGradient(fill, 15);
-console.log("New colors:", newColors);
+fills.forEach((fill, fillIndex) => {
+  if (isGradient(fill)) {
```

Comment:
Rather than nesting everything here, we can keep things a little simpler by early-returning when we don't have a gradient:
```js
if (!isGradient(fill) return;
```